File: sources/sources-1.0.0-m2/net.sf.j2s.java.org.eclipse.swt/src/org/eclipse/swt/internal/xhtml/TestTable.java
package org.eclipse.swt.internal.xhtml;

import java.util.Date;

import org.eclipse.swt.internal.RunnableCompatibility;

public class TestTable {

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		Element table = document.createElement("TABLE");
		table.style.position = "absolute";
		table.style.left = "100px";
		table.style.top = "25px";
		table.style.border = "1px solid red";
		table.style.backgroundColor = "white";
		document.body.appendChild(table);

		// Element thead = document.createElement("THEAD");
		// table.appendChild(thead);

		Element tbody = document.createElement("TBODY");
		table.appendChild(tbody);

		Date last = new Date();

		for (int i = 0; i < 100; i++) {
			Element tr = document.createElement("TR");
			tbody.appendChild(tr);

			for (int j = 0; j < 10; j++) {
				Element td = document.createElement("TD");
				tr.appendChild(td);

				Element div = document.createElement("DIV");
				td.appendChild(div);

				div.appendChild(document.createTextNode("Hi"
						+ Math.round(Math.random() * 20)));
//				div.onclick = new RunnableCompatibility() {
//
//					public void run() {
//						System.out.println("selected");
//					}
//
//				};
			}
		}
		System.out.println(new Date().getTime() - last.getTime());
	}

}

File: trunk/sources/net.sf.j2s.lib/j2slib/org/eclipse/swt/internal/xhtml/TestTable.js
$_J("org.eclipse.swt.internal.xhtml");
Clazz.load(null,"$wt.internal.xhtml.TestTable",["java.util.Date"],function(){
c$=$_C(function(){
$_Z(this,arguments);
},$wt.internal.xhtml,"TestTable");
c$.main=$_M(c$,"main",
function(args){
var table=d$.createElement("TABLE");
table.style.position="absolute";
table.style.left="100px";
table.style.top="25px";
table.style.border="1px solid red";
table.style.backgroundColor="white";
d$.body.appendChild(table);
var tbody=d$.createElement("TBODY");
table.appendChild(tbody);
var last=new java.util.Date();
for(var i=0;i<100;i++){
var tr=d$.createElement("TR");
tbody.appendChild(tr);
for(var j=0;j<10;j++){
var td=d$.createElement("TD");
tr.appendChild(td);
var div=d$.createElement("DIV");
td.appendChild(div);
div.appendChild(d$.createTextNode("Hi"+Math.round(Math.random()*20)));
}
}
System.out.println(new java.util.Date().getTime()-last.getTime());
},"~A");
});
